coursebook/week-2/dom-manipulation-ws.md

```diff
@@ -132,3 +132,324 @@ We can access _properties_ of DOM elements using javscript.
+
+  return introductionParagraph.textContent
+  
+```
+
+</p>
+</details>
+
+### Question 3
+
+<details><summary>Hint 1</summary>
+<p>
+So, in the previous question, we found out how to get the text content ('.textContent') of a DOM element. Now we simply need to check if a given string 'str' is included in that element - eg,
+
+``` javascript 
+
+"hereisastringofletters".includes("str") //returns true 
+
+```
+
+</p>
+</details>
+
+<details><summary>Solution</summary>
+<p>
+
+``` javascript
+
+var containsString = function (str) {
+
+  let introductionParagraph = document.getElementById('introduction');
+
+    if(introductionParagraph.textContent.includes(str)) {
+    return true
+  }
+  else {
+    return false
+  }
+
+}
+
+```
+
+</p>
+</details>
+
+### Question 4
+<details><summary>Hint 1</summary>
+<p>
+We need to retrieve the 'firstName' text input. If we look at the index.html file, we can see the <form> element, which has a 'value' query containing the text 'John':
+
+![form](https://i.imgur.com/8khxvgr.png)
+
+We can also see this in the browser window. It's the left-most box:
+
+![form-in-window](https://i.imgur.com/iHGCfvJ.png)
+
+First, we need to select the correct query selector. There are three <input> elements and we need to only select the one for 'firstName'. We can do this by using
+
+``` document.querySelector('input') //gets us the first input ``` 
+
+``` document.querySelector('input[name="firstName"]') //gets us the first input containing the name="firstName" property ```
+
+</p>
+
+</details>
+<details><summary>Solution</summary>
+<p>
+
+``` javascript
+
+var getFirstNameValue = function () {
+
+return document.querySelector('input[name="firstName"]').value;
+
+}
+
+```
+
+</p>
+</details>
+
+### Question 5
+<details><summary>Hint 1</summary>
+<p>
+Perhaps someone on the next cohort can write the hints to questions 5-9?
+</p>
+</details>
+<details><summary>Hint 2</summary>
+<p>
+
+</p>
+</details>
+<details><summary>Solution</summary>
+<p>
+
+```
+
+var getValue = function (inputName) {
+
+// Code here
+  return document.querySelector(`input[name="${inputName}"]`).value;
+
+}
+
+```
+
+</p>
+</details>
+
+### Question 6
+<details><summary>Hint 1</summary>
+<p>
+
+</p>
+</details>
+<details><summary>Hint 2</summary>
+<p>
+
+</p>
+</details>
+<details><summary>Solution</summary>
+<p>
+
+```
+
+var updateStateValue = function (formState, inputName) {
+
+// Code here
+  let clone = Object.assign({},formState);
+  clone[inputName]= getValue(inputName);
+  return clone;
+}
+
+```
+
+</p>
+</details>
+
+### Question 7
+<details><summary>Hint 1</summary>
+<p>
+
+</p>
+</details>
+<details><summary>Hint 2</summary>
+<p>
+
+</p>
+</details>
+<details><summary>Solution</summary>
+<p>
+
+```
+
+var updateStateValues = function (formState, inputNames) {
+
+// Code here
+  let clone = Object.assign({},formState);
+  inputNames.forEach(name=>clone[name]= getValue(name));
```

Comment:
arrow functions are usually bit advanced at this point, we have a workshop about JavaScript ES6 syntax (that includes arrow functions). so use a normal `function` here instead to avoid confusion.